synapse/api/auth.py

```diff
@@ -120,6 +134,42 @@ def check_joined_room(self, room_id, user_id, current_state=None):
         defer.returnValue(member)
 
     @defer.inlineCallbacks
+    def check_user_was_in_room(self, room_id, user_id, current_state=None):
+        """Check if the user was in the room at some point.
+        Args:
+            room_id(str): The room to check.
+            user_id(str): The user to check.
+            current_state(dict): Optional map of the current state of the room.
+                If provided then that map is used to check whether they are a
+                member of the room. Otherwise the current membership is
+                loaded from the database.
+        Raises:
+            AuthError if the user was never in the room.
+        Returns:
+            A deferred membership event for the user if the user was in
```

Comment:
It is unclear from this comment whether it will return the join event (from when the user *was* in the room) or the leave event, if the user has left the room.